ouroboros.go

```diff
@@ -132,20 +132,24 @@ func (o *Ouroboros) setupConnection() error {
 	o.muxer = muxer.New(o.conn)
 	// Start Goroutine to pass along errors from the muxer
 	go func() {
-		err, ok := <-o.muxer.ErrorChan
-		// Break out of goroutine if muxer's error channel is closed
-		if !ok {
+		select {
+		case <-o.doneChan:
 			return
+		case err, ok := <-o.muxer.ErrorChan:
+			// Break out of goroutine if muxer's error channel is closed
+			if !ok {
+				return
+			}
+			if errors.Is(err, io.EOF) || errors.Is(err, io.ErrUnexpectedEOF) {
+				// Return a bare io.EOF error if error is EOF/ErrUnexpectedEOF
+				o.ErrorChan <- io.EOF
+			} else {
+				// Wrap error message to denote it comes from the muxer
+				o.ErrorChan <- fmt.Errorf("muxer error: %s", err)
```

Comment:
It's an error to an error channel, skip error in the output.

Reply:
That's not new code (just moved around), but I purposely prefix `muxer error: ` on there so you can actually know where the error came from instead of just getting a random I/O related error without context